File: Zarina/TaZQL/src/ast/expression/comparison/GreaterEqual.java
package ast.expression.comparison;

import ast.expression.Binary;
import ast.expression.Expression;
import ast.expression.IExpressionVisitor;
import ast.type.ChoiceType;

// Greater than or equal to(>=)

public class GreaterEqual extends Binary {
				
	public GreaterEqual (Expression leftExp, Expression rightExp) {
		super(leftExp, rightExp);
	}
									
	@Override
	public <T> T accept(IExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}
	
	@Override
	public String toString() {
		return this.getLeftExpression().toString() + " >= " + this.getRightExpression().toString();
	}
	
	@Override
	public ChoiceType getType() {
		return new ChoiceType();
	}
}

File: guybas/QL/AST/Expressions/Operations/greater_equal.py
import QL.AST.Expressions.Operations.equal as e


class GreaterEqual(e.Equal):

    def set_string_operator(self):
        return " >= "

    # get the return _type of the _expression
    def return_type(self, type_map):
        return bool

    def eval(self, x, y):
        return x >= y